### PR TITLE
Cherry pick tooltip support to 0.63 branch

### PR DESCRIPTION
**Pull request by HeyImChris**
#### Please select one of the following
- [ ] I am removing an existing difference between facebook/react-native and microsoft/react-native-macos :thumbsup:
- [ ] I am cherry-picking a change from Facebook's react-native into microsoft/react-native-macos :thumbsup:
- [X] I am making a fix / change for the macOS implementation of react-native
- [X] I am making a change required for Microsoft usage of react-native

## Summary

This change was checked in to master with commit `308929130e426730a6cd0818c95a4cbf68a0bcc7`. Moving it to 0.63 so downstream stable dependencies can leverage it.

`git cherry-pick 308929130e426730a6cd0818c95a4cbf68a0bcc7`

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/react-native-macos/pull/713)

### CODE REVIEW

**Comment by harrieshin** [comment]
that's weird. I thought Amanda made multiple file changes... cherry picked wrong?

**Comment by chiuam** [comment]
> that's weird. I thought Amanda made multiple file changes... cherry picked wrong?

+1. The changes I made to the components for adding the tooltip prop.

**Comment by HeyImChris** [comment]
> > that's weird. I thought Amanda made multiple file changes... cherry picked wrong?
> 
> +1. The changes I made to the components for adding the tooltip prop.

Whoops yea something went wrong with the cherry pick. Let me try again

**Review by chiuam — APPROVED** [review]

**Review by harrieshin — APPROVED** [review]

**Review by Saadnajmi — APPROVED** [review]
